docs: TestContainers integration (#710)
qdrant/landing_page: qdrant-landing/content/documentation/frameworks/testcontainers.md

---
title: TestContainers
weight: 2700
---

# TestContainers

[TestContainers](https://testcontainers.com/) is an open source framework for providing throwaway, lightweight instances of databases, message brokers, web browsers, or just about anything that can run in a Docker container.

> Qdrant is available as a [TestContainers module](https://testcontainers.com/modules/qdrant/) in multiple languages facilitating the spawning of a Qdrant instance for end-to-end testing.

Reviewer: The link goes to a Testcontainers doc which specifies `v1.7.4`. That conflicts with v1.8.1 in several places, in this file.

Could we specify `latest` in both docs?

Author: https://github.com/qdrant/landing_page/pull/710/commits/8381fb3f31b36f0a49bf5de1187061d001ffb4d2

## Usage

```java
import org.testcontainers.qdrant.QdrantContainer;

QdrantContainer qdrantContainer = new QdrantContainer("qdrant/qdrant:v1.8.1");
```

```go
import (
    "github.com/testcontainers/testcontainers-go"
    "github.com/testcontainers/testcontainers-go/modules/qdrant"
)

qdrantContainer, err := qdrant.RunContainer(ctx, testcontainers.WithImage("qdrant/qdrant:v1.8.1"))
```

```typescript
import { QdrantContainer } from "@testcontainers/qdrant";

const qdrantContainer = await new QdrantContainer("qdrant/qdrant:v1.8.1").start();
```

```python
from testcontainers.qdrant import QdrantContainer

qdrant_container = QdrantContainer("qdrant/qdrant:v1.8.1").start()
```

> The container modules

Author: Sorry. Maybe I thought of the lines but didn't actually type them out.

> The container modules

Author: https://github.com/qdrant/landing_page/pull/710/commits/8381fb3f31b36f0a49bf5de1187061d001ffb4d2

---

> title: TestContainers

Reviewer 2: ```suggestion
title: Testcontainers
```

weight: 2700
---

> # TestContainers

Reviewer 2: ```suggestion
# Testcontainers
```

> [TestContainers](https://testcontainers.com/) is an open source framework for providing throwaway, lightweight instances of databases, message brokers, web browsers, or just about anything that can run in a Docker container.

Reviewer 2: ```suggestion
[Testcontainers](https://testcontainers.com/) is an open source framework for providing throwaway, lightweight instances of databases, message brokers, web browsers, or just about anything that can run in a Docker container.
```

> Qdrant is available as a [TestContainers module](https://testcontainers.com/modules/qdrant/) in multiple languages facilitating the spawning of a Qdrant instance for end-to-end testing.

Reviewer 2: ```suggestion
Qdrant is available as a [Testcontainers module](https://testcontainers.com/modules/qdrant/) in multiple languages facilitating the spawning of a Qdrant instance for end-to-end testing.
```

> [Testcontainers](https://testcontainers.com/) is an open source framework for providing throwaway, lightweight instances of databases, message brokers, web browsers, or just about anything that can run in a Docker container.
> 
> Qdrant is available as a [Testcontainers module](https://testcontainers.com/modules/qdrant/) in multiple languages facilitating the spawning of a Qdrant instance for end-to-end testing.

Reviewer: Since you're quoting directly from the Testcontainers website:

```suggestion
Qdrant is available as a [Testcontainers module](https://testcontainers.com/modules/qdrant/) in multiple languages. It  facilitates the spawning of a Qdrant instance for end-to-end testing.

As noted by [Testcontainers](https://testcontainers.com/), it "is an open source framework for providing throwaway, lightweight instances of databases, message brokers, web browsers, or just about anything that can run in a Docker container."
```

## Usage

```java
import org.testcontainers.qdrant.QdrantContainer;

QdrantContainer qdrantContainer = new QdrantContainer("qdrant/qdrant");
```

```go
import (
    "github.com/testcontainers/testcontainers-go"
    "github.com/testcontainers/testcontainers-go/modules/qdrant"
)

qdrantContainer, err := qdrant.RunContainer(ctx, testcontainers.WithImage("qdrant/qdrant"))
```

```typescript
import { QdrantContainer } from "@testcontainers/qdrant";

const qdrantContainer = await new QdrantContainer("qdrant/qdrant").start();
```

```python
from testcontainers.qdrant import QdrantContainer

qdrant_container = QdrantContainer("qdrant/qdrant").start()
```

> TestContainers modules provide options/methods to configure ENVs, volumes and virtually everything you can configure in a Docker container.

Reviewer: ```suggestion
Testcontainers modules provide options/methods to configure ENVs, volumes and virtually everything you can configure in a Docker container.
```

> Testcontainers modules provide options/methods to configure ENVs, volumes and virtually everything you can configure in a Docker container.

Reviewer: Fixing a grammar error

```suggestion
Testcontainers modules provide options/methods to configure ENVs, volumes, and virtually everything you can configure in a Docker container.
```

## Further reading

> - [TestContainers Guides](https://testcontainers.com/guides/)

Reviewer 2: ```suggestion
- [Testcontainers Guides](https://testcontainers.com/guides/)
```

> - [TesContainers Qdrant Module](https://testcontainers.com/modules/qdrant/)

Reviewer 2: ```suggestion
- [Tescontainers Qdrant Module](https://testcontainers.com/modules/qdrant/)
```

> - [Tescontainers Qdrant Module](https://testcontainers.com/modules/qdrant/)

Reviewer: ```suggestion
- [Testcontainers Qdrant Module](https://testcontainers.com/modules/qdrant/)
```
